frontend/test/metabase/scenarios/dashboard/x-rays.cy.spec.js

```diff
@@ -22,4 +27,56 @@ describe("scenarios > x-rays", () => {
     cy.findByText("Per state");
     cy.get(".Card").should("have.length", 11);
   });
+
+  it.skip("should work on questions with explicit joins (metabase#13112)", () => {
+    const PRODUCTS_ALIAS = "Products";
+
+    cy.request("POST", "/api/card", {
+      name: "13112",
+      dataset_query: {
+        type: "query",
+        query: {
+          "source-table": ORDERS_ID,
+          joins: [
+            {
+              fields: "all",
+              "source-table": PRODUCTS_ID,
+              condition: [
+                "=",
+                ["field-id", ORDERS.PRODUCT_ID],
+                ["joined-field", PRODUCTS_ALIAS, ["field-id", PRODUCTS.ID]],
+              ],
+              alias: PRODUCTS_ALIAS,
+            },
+          ],
+          aggregation: [["count"]],
+          breakout: [
+            ["datetime-field", ["field-id", ORDERS.CREATED_AT], "month"],
+            ["joined-field", PRODUCTS_ALIAS, ["field-id", PRODUCTS.CATEGORY]],
+          ],
+        },
+        database: 1,
+      },
+      display: "line",
+      visualization_settings: {},
+    }).then(({ body: { id: QUESTION_ID } }) => {
+      cy.server();
+      cy.route("POST", `/api/card/${QUESTION_ID}/query`).as("cardQuery");
+      cy.route("POST", "/api/dataset").as("dataset");
+
+      cy.visit(`/question/${QUESTION_ID}`);
+
+      cy.wait("@cardQuery");
+      cy.get(".dot")
+        .eq(23) // Random dot
+        .click({ force: true });
+      cy.findByText("X-ray").click();
+
+      cy.wait("@dataset");
```

Comment:
The tricky part here is that X-ray takes quite a long time. Waiting for dataset didn't time out for me locally, but I am a bit concerned how it will behave in CI. However, without it the test gives false positives, so we **have** to wait for it.

If CI times out in the future, I could maybe add some previous waits or additional checks to give `@dataset` some time.

Reply:
Yeah, 5 seconds might be cutting it a bit short on an overloaded CI - perhaps 30 seconds would prevent any flakes.
`cy.wait("@dataset", {timeout: 30000});` ?

Reply:
^ Yeah, that makes sense. It will wait **up to** 30s, so we're not losing anything.